indexer/src/indexer.ts

```diff
@@ -873,6 +874,198 @@ export const createIndexerService = (config: IndexerDbCredentials) => {
+             TotalPoolAggregates AS (SELECT ti.pool_address                                   AS pool_address,
+                                            ti.tvl_usd                                        AS total_liquidity_usd,
+                                            py.total_yield_usd                                AS total_swap_fees,
+                                            py.total_yield_usd * 0.1                          AS total_tower_swap_fees,
+                                            al.total_active_liquidity_positions               AS total_active_liquidity_positions,
+                                            al.token0_inflow_count + al.token1_inflow_count   AS total_liquidity_inflow,
+                                            al.token0_outflow_count + al.token1_outflow_count AS total_liquidity_outflow,
+                                            sd.total_distinct_wallet_swaps                    AS total_distinct_wallet_swaps,
+                                            sd.swaps_done                                     AS total_swaps_done,
+                                            id.total_incentives_done                          AS total_incentives_done
+                                     FROM TokenInfo ti
+                                              LEFT JOIN PoolYield py ON ti.pool_address = py.pool_address
+                                              LEFT JOIN ActiveLiquidityPositions al ON ti.pool_address = al.pool_address
+                                              LEFT JOIN SwapsDone sd ON ti.pool_address = sd.pool_address
+                                              LEFT JOIN IncentivesDone id ON ti.pool_address = id.pool_address
+                                     WHERE ti.pool_address = ${poolAddressesSql})
+        SELECT SUM(tpa.total_liquidity_usd)              AS total_liquidity_usd,
+               SUM(tpa.total_swap_fees)                  AS total_swap_fees,
+               SUM(tpa.total_tower_swap_fees)            AS total_tower_swap_fees,
+               SUM(tpa.total_active_liquidity_positions) AS total_active_liquidity_positions,
+               SUM(tpa.total_liquidity_inflow)           AS total_liquidity_inflow,
+               SUM(tpa.total_liquidity_outflow)          AS total_liquidity_outflow,
+               SUM(tpa.total_distinct_wallet_swaps)      AS total_distinct_wallet_swaps,
+               SUM(tpa.total_swaps_done)                 AS total_swaps_done,
+               SUM(tpa.total_incentives_done)            AS total_incentives_done,
+               ${startHeight}                            AS start_height,
+               ${endHeight}                              AS end_height
+        FROM TotalPoolAggregates tpa;
+    `;
+
+    try {
+      const result = await client.execute(query);
+
+      if (result.rows.length === 0) {
+        throw new Error("No data found for the given date range.");
+      }
+
+      return result.rows[0] as AggregatedMetrics;
```

Comment:
what happens to rows[1] etc? Should this be a reduce or something similar?

Reply:
Yes, can do, we aggregate all the individual pool totals, so there should only ever be one row actually.

Reply:
As discussed